store/postgres/src/store.rs

```diff
@@ -615,25 +603,32 @@ impl StoreTrait for Store {
             panic!("transact_block_operations must transact a single block only");
         }
 
+        // All operations should apply only to entities in this subgraph
+        for op in &operations {
+            if op.entity_key().subgraph_id != subgraph_id {
+                panic!("transact_block_operations must affect only entities in the subgraph");
+            }
+        }
+
         // Fold the operations of each entity into a single one
         let operations = EntityOperation::fold(&operations);
 
         let conn = self.conn.lock().unwrap();
 
-        conn.transaction::<(), _, _>(|| {
-            self.apply_entity_operations(&*conn, operations, block_ptr_to)?;
+        conn.transaction(|| {
+            let event_source = EventSource::EthereumBlock(block_ptr_to);
+            self.apply_entity_operations_with_conn(&*conn, operations, event_source)?;
             self.update_subgraph_block_pointer(&*conn, subgraph_id, block_ptr_from, block_ptr_to)
         })
     }
 
-    /// Sets an entity.
-    fn apply_set_operation(
+    fn apply_entity_operations(
         &self,
-        operation: EntityOperation,
-        op_event_source: String,
+        operations: Vec<EntityOperation>,
+        event_source: EventSource,
     ) -> Result<(), Error> {
         let conn = self.conn.lock().unwrap();
-        self.apply_set_operation_with_conn(&conn, operation, &op_event_source)
+        conn.transaction(|| self.apply_entity_operations_with_conn(&conn, operations, event_source))
```

Comment:
What's the effect of this extra transaction?

Reply:
Makes sense, were we failing to do this before or was this in another part of the code?

Reply:
Wasn't being done before, although less of an issue back then because this function used to only commit one entity change